### PR TITLE
Getter/setter compatibility

### PR DESCRIPTION
**Pull request by akauppi**
Current Leaflet `L.Class.extend` does not work well if the derived class uses ECMAscript 5 get/set methods. This PR fixed that, based on [this blog from 2007](http://ejohn.org/blog/javascript-getters-and-setters/).

Why does this matter?

Getters/setters seem like a nice addition to JavaScript and the Leaflet class extension mechanism can be made to be compatible with them.

There are ways to bypass the current limitation (see [SO](http://stackoverflow.com/questions/30197434/javascript-getter-why-does-it-get-called-at-construction)) but those are kind of clumsy, causing get/set code to end up separated from normal methods.

Are there downsides?

- the suggested implementation does not check for browser capabilities. This should be added. Older Leaflet-supported browsers would not know of getters/setters. But this is doable.

- `jake test` now expects support for the get/set syntax from the underlying V8 engine. It works for mine (node 0.12.2) but I'm not sure about older node versions. Don't think this would be a major concern, though.

To sum up. This PR allows the library user to use `L.Class.extend` mechanism, together with his/her own code utilizing ECMAscript 5 getters/setters in his/her own code. It does not suggest Leaflet should use getters anywhere. I'd like to hear comments on the value of such a change - i.e. have others come across this current limitation.

### CODE REVIEW

**Comment by mourner** [comment]
Thanks for the PR! While it is certainly useful in some cases, my main concern about this change is performance. `L.extend` is very critical function, it's used throughout the whole Leaflet code in many places, and takes up considerable time when dealing with many layers according to profiling results. This change will hit performance a lot, while satisfying a very rare use case, so it's something I can't afford in the core.

However, this can easily be released as a plugin! It would override `L.Util.extend`, accomplishing the same.

**Comment by akauppi** [comment]
Sure, @mourner. 

I mostly just wanted to "report" the surprise that getters/setters didn't work.

Also I was worried about the performance implications - and I tremendously value you keeping a keen eye on those.

However, if you ever wish to have support for them, there's another way (came to think of it after filing the PR). One could have `.props` similar to how `.statics`, `.includes` etc. currently are - i.e. have the properties cut down to their own area, which should help with the performance aspect, yet fit nicely (I think) to the overall Leaflet way.

For now, nothing needs to be done.
